src/charm.py

```diff
@@ -84,18 +75,18 @@ def generate_worker_layer(self, worker: Worker) -> Layer:
 
         This method assumes that worker.roles is valid.
         """
+
         roles = worker.roles
         if not len(roles) == 1:
             raise RolesConfigurationError(
                 f"Worker can only have 1 role configured. {len(roles)} found."
             )
         role = roles[0]
-        if role not in self._valid_roles:
```

Comment:
why did you remove this error?

Reply:
Not needed anymore after we've changed the roles config options to be `role-X: boolean`